### PR TITLE
tools-osx (new formula)

### PR DESCRIPTION
**Pull request by jawshooah**

### CODE REVIEW

File: Library/Formula/tools-osx.rb [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  conflicts_with 'trash', :because => 'both install a trash binary'
+
+  def install
+    rake
+    libexec.install 'bin'
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    system "#{bin}/trash"
```
Comment (jawshooah) [inline Library/Formula/tools-osx.rb]: Done in 6a4b2ba.

**Comment by jawshooah** [comment]
Shall I squash the changes to `tools-osx.rb` into one commit?

File: Library/Formula/tools-osx.rb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  conflicts_with "trash", :because => "both install a trash binary"
+
+  def install
+    rake
+    libexec.install "bin"
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+  end
+
+  test do
+    system "/usr/bin/touch", "testfile"
```
Comment (DomT4) [inline Library/Formula/tools-osx.rb]: You can just do ` touch "testfile" ` here. No need to invoke the system for this line :).

Comment (DomT4) [inline Library/Formula/tools-osx.rb]: (Otherwise, looks great. Thanks for taking the time to fix the issues!)

File: Library/Formula/tools-osx.rb [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class ToolsOsx < Formula
+  homepage 'https://github.com/morgant/tools-osx'
```
Comment (DomT4) [inline Library/Formula/tools-osx.rb]: Thanks!

**Comment by DomT4** [comment]
Yeah, if you squash the changes to the ` tools-osx.rb  ` formula into one commit, that would be great :+1:. Obviously the other formulae changes in the PR can remain separate commits.

**Comment by jawshooah** [comment]
@DomT4 Made the edits you suggested, and squashed changes to `tools-osx.rb`. Thanks!

**Comment by jawshooah** [comment]
It's worth noting that the version in the `url` is specifically for the `trash` binary in the `tools-osx` suite. The maintainer has been bumping version numbers for binaries individually, instead of for the suite as a whole. How can we work around this?

**Comment by DomT4** [comment]
It looks like the latest release, the one you've tagged, contains all of the tools, even if he has named it otherwise. Doing ` rake ` in the top level dir produces a bin directory with:

```
├── 32bitSafePerl
├── clipcat
├── eject
├── ql
├── swuser
├── trash
├── urlmnt
└── with
```

So theoretically:

```ruby
def install
  prefix.install_metafiles
  libexec.install "bin"
  bin.write_exec_script Dir["#{libexec}/bin/*"]
end
```

Should work, and install all the tools?

**Comment by jawshooah** [comment]
Right, it installs all the tools just fine. I was more concerned with version tracking.

**Comment by DomT4** [comment]
Oh I see. Apologies. Misunderstanding there. With regards to versioning, we can always set the version to the last release tarball, and then whatever comes next can follow the same pattern, and unless he releases in the past the issue should stay fixed. Something like ` version "2012-12-01" ` or something might work.

We do have a slight problem though:

```
clipcat
/usr/local/bin/32bitSafePerl: line 69: : command not found
```

Running the 32bitSafePerl script directly hits:
```
32bitSafePerl
/usr/local/bin/32bitSafePerl: line 65: /tmp/32bitSafePerl/: Is a directory
/usr/local/bin/32bitSafePerl: line 69: : command not found
rm: /tmp/32bitSafePerl/: is a directory
```

:cry:

**Comment by DomT4** [comment]
It seems to be passing the CI though, so theoretically, there's nothing wrong with it, theoretically. Does it all run silky-smooth for you locally?

**Comment by jawshooah** [comment]
Ah, damn. Seems like `32bitSafePerl` requires Lion or earlier. I get the same error running the `with` binary. Also seeing a problem with `urlmnt`:
```/usr/local/bin/urlmnt: line 27: print: command not found```
Is there a way to add `depends_on :macos` with a _maximum_ release?

**Comment by jawshooah** [comment]
Since the maintainer has disabled submitting issues on the repo, I'll ping him here. @morgant, are there any plans to support 10.8 and later?

**Comment by DomT4** [comment]
Yeah, depends_on MaximumMacOSRequirement => :lion

That should do the trick.

**Comment by jawshooah** [comment]
Thanks, added the dependency. There is an additional tool, `dict`, that is only installed by the Rakefile if MacRuby is installed. Does that warrant a `caveats` section?

**Comment by DomT4** [comment]
To be honest, I'd be tempted to remove the dict even if it is created and see if people request its return later. You can do this with 'if bin/'dict' exist? rm bin/'dict' end. Obviously, format it as expected and not how I had to type it so the email reply wouldn't break ;).

**Comment by jawshooah** [comment]
Probably a good idea if we're looking for consistency. Done.

**Comment by jawshooah** [comment]
Well, after taking a closer look at `urlmnt`, it seems it actually does nothing at all for the moment.

This is obviously not an official release, and the most recent one is from the Snow Leopard era, so I'm just going to close this for now. Hopefully we'll see an update soon.

**Comment by morgant** [comment]
I know this is closed at the moment, but I'll chime in on some of the issues mentioned above and will work on a new release since `2011-06-02` is still the latest stable build:

* Yes, for better or worse, I had been tagging for both releases and stable versions of the individual tools (this is a relic of the days when it was an SVN repo and helps me find those stable points in the history).
* `urlmnt` is, indeed, a stub and can be ignored.
* `32bitSafePerl` was a workaround for supporting `with` (required MacPerl which is--or was--32-bit only) on Lion which no-longer supported 32-bit Perl, though it required it. This may be resolved as easily as changing a version check from = to >=, but I'll have to test on 10.8-10.10.
* `dict` and it's requirement of MacRuby is actually one of the reasons we never got to a stable release as @dpk and I mucked around with various ways to optionally support/install it. I've considered splitting it out into it's own repo. I'll check in with @dpk about that since it was his submission.

Let me know if you have any further questions.

**Comment by jawshooah** [comment]
Thanks for the speedy response! I'll keep an eye out for a stable release and reopen this when one is ready.

**Pull request closed without merging**